Sources/Socket.swift

```diff
@@ -2563,10 +2586,6 @@ public class Socket: SocketReader, SocketWriter {
 	/// 	- bufSize: 	The size of the buffer.
 	///		- address: 	Address to write data to.
 	///
-	/// - Throws: `Socket.SOCKET_ERR_RECV_BUFFER_TOO_SMALL` if the buffer provided is too small.
-	///		Call again with proper buffer size (see `Error.bufferSizeNeeded`) or
-	///		use `readData(data: NSMutableData)`.
-	///
```

Comment:
You removed this and the functionality...  `readDataIntoStorage` reads the data into an *internal* buffer.  Then, upon return to this function, the data read is transferred to the caller's buffer.  However, if the buffer is too small, an exception is thrown that contains the required buffer size.  Since the data is still in the internal storage buffer, the caller can still call again and get the complete message.  There's no reason to omit this functionality just because it's UDP.